javatests/arcs/core/storage/StorageKeyFactoryTest.kt

```diff
@@ -13,9 +13,27 @@ import org.junit.runners.JUnit4
 @RunWith(JUnit4::class)
 class StorageKeyFactoryTest {
 
+    @Test
+    fun backingStore_namedEntity() {
+        val namedSchema = Schema(setOf(SchemaName("Foo")),
+            SchemaFields(emptyMap(), emptyMap()),
+            "SomeHash"
+        )
+        val arcId = ArcId.newForTest("myId")
+
+        val storeOptionsEmptyName = StorageKeyFactory.BackingStorageKeyOptions(
+            arcId, namedSchema
+        )
+
+        assertThat(storeOptionsEmptyName.unique).isEqualTo(namedSchema.name?.name)
```

Comment:
maybe instead use `requireNotNull(namedSchema.name).name`, or even just "Foo" here, to avoid a double bug when it's empty in both :)